hasher-matcher-actioner/hmalib/common/s3_adapters.py

```diff
@@ -209,37 +210,52 @@ class ThreatUpdateS3Store(tu.ThreatUpdatesStore):
     ThreatUpdatesStore, but stores files in S3 instead of local filesystem.
     """
 
+    CHECKPOINT_SUFFIX = ".checkpoint"
+
     def __init__(
         self,
         privacy_group: int,
         app_id: int,
-        s3_bucket: Bucket,  # Not typable?
+        s3_client: S3Client,
+        s3_bucket_name: str,
         s3_te_data_folder: str,
         data_store_table: str,
         supported_signal_types: t.List[SignalType],
     ) -> None:
         super().__init__(privacy_group)
         self.app_id = app_id
         self._cached_state: t.Optional[t.Dict] = None
-        self.s3_bucket = s3_bucket
         self.s3_te_data_folder = s3_te_data_folder
         self.data_store_table = data_store_table
         self.supported_indicator_types = self._get_supported_indicator_types(
             supported_signal_types
         )
+        self.s3_client = s3_client
+        self.s3_bucket_name = s3_bucket_name
+
+    @classmethod
+    def indicator_type_str_from_signal_type(
+        cls, signal_type: t.Type[SignalType]
+    ) -> str:
+        """
+        This mapping is only necessary for types that are in the process of
+        being migrated. eg. VideoMD5.
+        """
+        if signal_type == VideoMD5Signal:
+            return "HASH_VIDEO_MD5"
```

Comment:
nit: I think we're going to forget about this here. Can we address this in py threatexchange?

Reply:
We must. But I do not have a clean solution. Partners need both HASH_MD5 and HASH_VIDEO_MD5 to be INDICATOR_TYPEs for VideoMD5 depending on:
a) whether they have started using threat_updates
b) their particular collaboration has been backfilled